sdk/messaging/azservicebus/internal/exported/error.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import "fmt"
+
+// Code is an error code, usable by consuming code to work with
+// programatically.
+type Code string
+
+const (
+	// CodeConnectionLost means our connection was lost and all retry attempts failed.
+	// This typicall reflects an extended outage or connection disruption and may
+	// require manual intervention.
+	CodeConnectionLost Code = "connlost"
+
+	// CodeLockLost means that the lock token you have for a message has expired.
+	// This message will be available again after the lock period expires, or, potentially
+	// go to the dead letter queue if delivery attempts have been exceeded.
+	CodeLockLost Code = "locklost"
+)
+
+// Error represents a Service Bus specific error.
+type Error struct {
+	// Code is a stable error code which can be used as part of programatic error handling.
+	// The codes can expand in the future, but the values (and their meaning) will remain the same.
+	Code     Code
+	innerErr error
+}
+
+// Error is an error message containing the code and a user friendly message, if any.
+func (e Error) Error() string {
+	return fmt.Sprintf("(%s): %s", e.Code, e.innerErr.Error())
```

Comment:
The usage of this is restricted to only a few spots, all of which only create this error because they're intentionally wrapping an error they classified. So in this case it's on the caller to do it correctly but they'd have to go out of their way to not do so.

Maybe there's a better way. I didn't want to end up with `NewError(err innerErr) (Error, error)` for instance, which would at least allow me to do the validation in the constructor but that's the only way I can see preventing this potential problem.

Reply:
For now I'll just do a check with nil and print out unknown error (just so it's easy to search and find should we get bug reports)

Reply:
Fixed! Also added a couple of tests, let me know if it's what you were thinking of.

Reply:
Unwrap() has now been removed.